packages/server/src/adapters/fetch/content-type/json/index.ts

```diff
@@ -16,79 +17,74 @@ export interface FetchHTTPContentTypeHandler<TRouter extends AnyRouter>
 
 export const getFetchHTTPJSONContentTypeHandler: <
   TRouter extends AnyRouter,
->() => FetchHTTPContentTypeHandler<TRouter> = () => ({
-  name: 'fetch-json',
-  isMatch: (headers) => {
-    return !!headers.get('content-type')?.startsWith('application/json');
-  },
-  getInputs: async (opts, info) => {
-    async function getRawProcedureInputOrThrow() {
-      const { req } = opts;
+>() => FetchHTTPContentTypeHandler<TRouter> = () => {
+  const cache = createConcurrentCache();
```

Comment:
Couldn't this just be 

```suggestion
  const getRawProcedureInputOrThrow = memoize(async () => {
    // ...impl
  })
```

it's more straight-forward than the concurrent cache function where the key value etc

it's easy to write a memo-fn

```ts

function memoize<TReturn>(fn: () => TReturn): () => TReturn {
  let cache: TReturn | undefined;
  return () => {
    if (!cache) {
      cache = fn();
    }
    return cache;
  };
}
```

Reply:
How would that work for individual batch numbers?

Reply:
We don't need to memoize the deserialization, do we? I

Reply:
So it won’t explode, but it is unnecessary work and getRawInput might be called multiple times depending on setup. Particularly for larger batches it’s a low hanging optimisation (I really just moved it inside here from resolveHTTPResponse to simplify that method)

I like the memo syntax you’re suggesting btw, and we could probably do it all in a more raw way